### PR TITLE
Tmds.DBus.Tool: Add an option to generate public types

### PR DESCRIPTION
**Pull request by matdug**
Currently the types that are generated by the codegen tool have an implicit "internal" access modifier.  This is fine if you intend to use the generated types from within the same assembly where they will be included, but they are not visible to other assemblies.
I would like to add the possibility to generate these types with the "public" access modifier so that they can be referenced from other assemblies. For this I propose the --public-types command line option.  The default behaviour when the option is omitted is for the types to remain "internal".

### CODE REVIEW

File: src/Tmds.DBus.Tool/CodeGenCommand.cs [hunk 1]
```diff
@@ -32,12 +34,13 @@ public override void Configure()
...
+            _noInternalsVisibleToOption = Configuration.Option("--no-ivt", "Don't add the InternalsVisibleTo Attribute", CommandOptionType.NoValue);
             _namespaceOption = Configuration.Option("--namespace", "C# namespace (default: <service>)", CommandOptionType.SingleValue);
             _outputOption = Configuration.Option("--output", "File to write (default: <namespace>.cs)", CommandOptionType.SingleValue);
             _catOption = Configuration.Option("--cat", "Write to standard out instead of file", CommandOptionType.NoValue);
             _skipOptions = Configuration.Option("--skip", "DBus interfaces to skip", CommandOptionType.MultipleValue);
             _interfaceOptions = Configuration.Option("--interface", "DBus interfaces to include, optionally specify a name (e.g. 'org.freedesktop.NetworkManager.Device.Wired:WiredDevice')", CommandOptionType.MultipleValue);
+            _publicTypesOption = Configuration.Option("--public-types", "The types generated by this tool will have a public access modifier (default: internal)", CommandOptionType.NoValue);
```
Comment (tmds) [inline src/Tmds.DBus.Tool/CodeGenCommand.cs:43]: nit: rename `--public-types` to `--public`
and `The code generated by this tool will have public access modifier (default: internal).`

**Comment by tmds** [comment]
@matdug thanks for this contribution! It looks good to me. I have a small suggestion on naming/phrasing.

**Comment by matdug** [comment]
Hi @tmds , I made the changes as per your comments.
Thanks,
Mathieu

**Comment by tmds** [comment]
Thank you @matdug!

**Comment by tmds** [comment]
@matdug this is part of v0.9.0 which I just uploaded to nuget.org.

**Pull request merged**